src/protagonist/API/Features/NamedQueries/Requests/CreateNamedQuery.cs

```diff
@@ -0,0 +1,63 @@
+﻿using API.Infrastructure.Requests;
+using DLCS.Core;
+using DLCS.Model.Assets.NamedQueries;
+using DLCS.Repository;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+
+namespace API.Features.NamedQueries.Requests;
+
+public class CreateNamedQuery : IRequest<ModifyEntityResult<NamedQuery>>
+{
+    public int CustomerId { get; }
+    
+    public NamedQuery NamedQuery { get; }
+    
+    public CreateNamedQuery(int customerId, NamedQuery namedQuery)
+    {
+        CustomerId = customerId;
+        NamedQuery = namedQuery;
+    }
+}
+
+public class CreateNamedQueryHandler : IRequestHandler<CreateNamedQuery, ModifyEntityResult<NamedQuery>>
+{  
+    private readonly DlcsContext dbContext;
+
+    public CreateNamedQueryHandler(DlcsContext dbContext)
+    {
+        this.dbContext = dbContext;
+    }
+    
+    public async Task<ModifyEntityResult<NamedQuery>> Handle(CreateNamedQuery request, CancellationToken cancellationToken)
+    {
+        if (request.CustomerId != request.NamedQuery.Customer)
+        {
+            return ModifyEntityResult<NamedQuery>.Failure("The user id provided in the named query model does not match the calling user's id",
```

Comment:
Could this ever happen? If so message should be `"The customer id... the calling customers id"` rather than ``"The user id.. the calling users id"``.

Reply:
Changed in eaf8c546ac8818e09d3dab9f99177bbabf4cf650 and 56c9807ec49448c634ee0f4faa0de8cd48c1bcc2